legacy/firmware/fsm_msg_ethereum.h

```diff
@@ -93,8 +92,25 @@ void fsm_msgEthereumSignTx(const EthereumSignTx *msg) {
 
   CHECK_PIN
 
-  if (!fsm_ethereumCheckPath(msg->address_n_count, msg->address_n, false,
-                             msg->chain_id)) {
+  const EthereumDefinitions_encoded_network_t *encoded_network = NULL;
+  const EthereumDefinitions_encoded_token_t *encoded_token = NULL;
+  if (msg->has_definitions) {
+    if (msg->definitions.has_encoded_network) {
+      encoded_network = (const EthereumDefinitions_encoded_network_t *)&msg
+                            ->definitions.encoded_network;
+    }
+    if (msg->definitions.has_encoded_token) {
+      encoded_token = (const EthereumDefinitions_encoded_token_t *)&msg
+                          ->definitions.encoded_token;
+    }
+  }
+
+  const EthereumDefinitionsDecoded *defs = get_EthereumDefinitionsDecoded(
```

Comment:
this whole block is big enough (and repeated below) to warrant its own helper with this signature:
```
const EthereumDefinitionsDecoded * get_definitions(bool has_definitions, const EthereumDefinitions *definitions, uint64_t chain_id, const char *to);
```

Reply:
9556b8f18d0dab6fc84e20f31bad8ba707737562